### ISSUE TITLE
unsupported compression method 188

### ISSUE THREAD

**Issue by schalekamp** [IMSMWU/RClickhouse#85]
`> con <- DBI::dbConnect(RClickhouse::clickhouse(), host="myhost", port=9001L, user="", password="")
> 
> res <- DBI::dbGetQuery(con,"select * from tbl_sample limit 10")
Error in select(conn@ptr, statement) : unsupported compression method 188`

I'm getting this error when I try to fetch data from my clickhouse DB.

**Comment by schalekamp** [comment]
Is there a way to set the compression method via the dbConnect call ?

**Comment by inkrement** [comment]
yes, there should be a "compression" argument with the default lz4.

**Comment by schalekamp** [comment]
when I run

`SELECT getSetting('network_compression_method');`

I get LZ4

Which should be supported by this library?

**Comment by inkrement** [comment]
Which version do you use for the server?

**Comment by schalekamp** [comment]
ClickHouse 21.9.2.17

**Comment by den-crane** [comment]
Invalid deserialization of Native format - expecting a compressed block not where it starts?

Maybe related to DateTime64 https://github.com/IMSMWU/RClickhouse/issues/84
Maybe related to LowCardinality(String). (see Clickhouse setting low_cardinality_allow_in_native_format)

**Comment by schalekamp** [comment]
I use both DateTime64 and LowCardinality(String) in my DDL

**Comment by inkrement** [comment]
Indeed, this is likely the problem at hand. We'll take a closer look at the issue and try to update the client accordingly. However, we cannot provide a time window for the next version.

**Comment by schalekamp** [comment]
Understood. Already a big thank you for taking the time to look into this.

**Comment by spgarbet** [comment]
This is Heisenbugged on my computer. Occasionally it will work, otherwise the reported compression type is a value taken from random memory. If I run it several times eventually it gives "fail to send data: Broken pipe". It's more likely to happen on queries that have longer results. 

```
Error in select(conn@ptr, statement) : unimplemented 1775107
> dbGetQuery(client, "SELECT COUNT(*) FROM historical.quotes")
Error in select(conn@ptr, statement) : unimplemented 11861
> dbGetQuery(client, "SELECT COUNT(*) FROM historical.quotes")
Error in select(conn@ptr, statement) : unimplemented 122
> dbGetQuery(client, "SELECT COUNT(*) FROM historical.quotes")
Error in select(conn@ptr, statement) : unimplemented 124
> dbGetQuery(client, "SELECT COUNT(*) FROM historical.quotes")
Error in select(conn@ptr, statement) : unimplemented 1152565
> dbGetQuery(client, "SELECT COUNT(*) FROM historical.quotes")
Error in select(conn@ptr, statement) : fail to send data: Broken pipe
```

It resets and works on a new connection. Once the error occurs it never works again.
